atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1695,12 +1697,15 @@ public void abort() {
                     throwIfPreCommitRequirementsNotMet(null, getStartTimestamp());
                 }
                 transactionOutcomeMetrics.markAbort();
-                if (perfLogger.isDebugEnabled()) {
+                if (transactionLengthLogger.isDebugEnabled()) {
                     long transactionMillis = TimeUnit.NANOSECONDS.toMillis(transactionTimerContext.stop());
-                    perfLogger.debug(
-                            "Aborted transaction {} in {} ms",
-                            SafeArg.of("startTimestamp", getStartTimestamp()),
-                            SafeArg.of("transactionTimeMillis", transactionMillis));
+
+                    if (transactionMillis > TXN_LENGTH_THRESHOLD) {
+                        transactionLengthLogger.debug(
+                                "Aborted transaction {} in {} ms",
+                                SafeArg.of("startTimestamp", getStartTimestamp()),
+                                SafeArg.of("transactionTimeMillis", transactionMillis));
```

Comment:
can we use a Duration here for readability? Could be as an additional arg if we don't want to break existing analyses.